src/main/java/net/imagej/ops/image/watershed/WatershedSeeded.java

```diff
@@ -0,0 +1,404 @@
+			neighborVoxels.clear();
+
+			// iterate the neighborhood of the pixel
+			raNeigh.setPosition(pos);
+			final Cursor<T> neighborhood = raNeigh.get().cursor();
+			while (neighborhood.hasNext()) {
+				neighborhood.fwd();
+				// Unlabeled neighbors go into the queue if they are not there
+				// yet
+				raOut.setPosition(neighborhood);
+				raMask.setPosition(raOut);
+				if (!raOut.get().isEmpty() && !raOut.isOutOfBounds()) {
+					final Integer label = raOut.get().iterator().next();
+					if (label == INIT && raMask.get().get()) {
+						raIn.setPosition(neighborhood);
+						neighborVoxels.add(new WatershedVoxel(IntervalIndexer.positionToIndex(raIn, out),
+								raIn.get().getRealDouble()));
+					} else {
+						if (label > WSHED && (!drawWatersheds || !neighborLabels.contains(label))) {
+							// store labels of neighbors in a list
+							neighborLabels.add(label);
+						}
+					}
+				}
+
+			}
+
+			if (drawWatersheds) {
+				// if the neighbors of the extracted voxel that have already
+				// been labeled
+				// all have the same label, then the voxel is labeled with their
+				// label.
+				raOut.setPosition(pos);
+				raOut.get().clear();
+				if (neighborLabels.size() == 1) {
+					raOut.get().add(neighborLabels.get(0));
+					// now that we know the voxel is labeled, add neighbors to
+					// list
+					for (final WatershedVoxel v : neighborVoxels) {
+						IntervalIndexer.indexToPosition(v.getPos(), out, raOut);
+						raOut.get().clear();
+						raOut.get().add(INQUEUE);
+						pq.add(v);
+					}
+				} else if (neighborLabels.size() > 1)
+					raOut.get().add(WSHED);
+			} else {
+				if (neighborLabels.size() > 0) {
+					raOut.setPosition(pos);
+					raOut.get().clear();
+
+					// take the label which most of the neighbors have
+					if (neighborLabels.size() > 2) {
+						final Map<Integer, Long> countLabels = neighborLabels.stream()
+								.collect(Collectors.groupingBy(e -> e, Collectors.counting()));
+						final Integer keyMax = Collections
+								.max(countLabels.entrySet(), Comparator.comparingLong(Map.Entry::getValue)).getKey();
+						raOut.get().add(keyMax);
+					} else {
+						raOut.get().add(neighborLabels.get(0));
+					}
+					// now that we know the voxel is labeled, add neighbors to
+					// list
+					for (final WatershedVoxel v : neighborVoxels) {
+						IntervalIndexer.indexToPosition(v.getPos(), out, raOut);
+						raOut.get().clear();
+						raOut.get().add(INQUEUE);
+						pq.add(v);
+					}
+				}
+			}
+		}
+
+		/*
+		 * Merge already present labels before calculation of watershed
+		 */
+		if (out() != null) {
+			final Cursor<LabelingType<Integer>> cursor = out().cursor();
+			while (cursor.hasNext()) {
+				cursor.fwd();
+				raOut.setPosition(cursor);
+				final List<Integer> labels = new ArrayList<>();
+				cursor.get().iterator().forEachRemaining(labels::add);
+				raOut.get().addAll(labels);
+			}
+		}
+
+	}
+
+	@Override
+	public boolean conforms() {
+		boolean conformed = true;
```

Comment:
I suggest that we require `mask` and `seeds` to have the same dimensionality as `in()`